### PR TITLE
add gradient descent example

### PR DESCRIPTION
**Pull request by mateogianolio**
Following @lucidrains neural network example, I took the liberty of adding a javascript version of gradient descent from part 2 in the series: [A Neural Network in 13 lines of Python (Part 2 - Gradient Descent)](https://iamtrask.github.io//2015/07/27/python-network-part2/)

### CODE REVIEW

**Comment by mateogianolio** [comment]
#45

**Comment by lucidrains** [comment]
haha nice, maybe i'll tackle this next with vectorious https://iamtrask.github.io/2016/02/25/deepminds-neural-stack-machine/ :D
